integrated_channels/blackboard/client.py

```diff
@@ -45,11 +51,232 @@ def update_content_metadata(self, serialized_data):
+                course_id=course_id,
+                column_id=column_id,
+                user_id=user_id,
+            ),
+        )
+
+    def _get(self, url, data=None):
+        """
+        Returns request's get response and raises Client Errors if appropriate.
+        """
+        get_response = self.session.get(url, params=data)
+        if get_response.status_code >= 400:
+            raise ClientError(get_response.text, get_response.status_code)
+        return get_response
+
+    def _patch(self, url, data):
+        """
+        Returns request's patch response and raises Client Errors if appropriate.
+        """
+        patch_response = self.session.patch(url, json=data)
+        if patch_response.status_code >= 400:
+            raise ClientError(patch_response.text, patch_response.status_code)
+        return patch_response
+
+    def _post(self, url, data):
+        """
+        Returns request's post response and raises Client Errors if appropriate.
+        """
+        post_response = self.session.post(url, json=data)
+        if post_response.status_code >= 400:
+            raise ClientError(post_response.text, post_response.status_code)
+        return post_response
+
+    def _get_courses_by_internal_id(self, internal_id):
+        """
+        Helper method to retrieve a course by it's Blackboard internal ID.
+        """
+        params = {'externalId': internal_id}
+
+        courses_response = self._get(self.generate_courses_url(), params).json()
+        return courses_response.get('results')
+
+    def _get_bb_user_id_from_enrollments(self, user_id, course_id):
+        """
+        Helper method to retrieve a user's Blackboard ID from a list of enrollments in a
+        Blackboard class.
+
+        Parameters:
+        -----------
+            user_id (str): The shared email of the user for both Blackboard and edX
+            course_id (str): The Blackboard course ID of which to search enrollments.
+        """
+        enrollments_response = self._get(self.generate_enrollment_url(course_id)).json()
+        for enrollment in enrollments_response.get('results'):
+            # No point in checking non-students
+            if enrollment.get('courseRoleId') == 'Student':
+                contact = enrollment.get('user').get('contact')
+                if contact.get('email') == user_id:
+                    return enrollment.get('userId')
+        raise ClientError(
+            'Could not find user={} enrolled in Blackboard course={}'.format(user_id, course_id),
+            HTTPStatus.NOT_FOUND.value
+        )
+
+    def _get_or_create_integrated_grade_column(self, course_id):
+        """
+        Helper method to search an edX integrated Blackboard course for the designated edX grade column.
+        If the column does not yet exist within the course, create it.
+
+        Parameters:
+        -----------
+            course_id (str): The Blackboard course ID in which to search for the edX final grade,
```

Comment:
since there may be both course_ids in play in our code, perhaps best to use prefix like bb_course_id or blackboard_course_id ?